org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -291,6 +320,9 @@ protected void copyPO(PO copy)
 	/** Immutable flag **/
 	private boolean m_isImmutable = false;
 
+	/** Record UUID **/
+	private String m_uuid = null;
```

Comment:
I think it should be better if we remove this instance variable as well. the uuid can be parameter that's being pass around. If you are keeping this, you will have to ensure it always has the right uuid value, otherwise the PO state can be inconsistent at time.

Reply:
@hengsin I tried to do this but looks get issue in get_WhereClause method as it invoked from many places and we have to pass null arguments on most of place. I think as currently the way it used without having getters and setters for m_uuid, we don't needs to maintain state of it. If we try to load value in it after reading record from DB or from RS, it may create more confusion as then after record always loaded by UU even ID present.

Reply:
I don't see why get_WhereClause is a problem, with get_WhereClause (boolean withValues, string uuid) and get_WhereClause (boolean withValues) { return get_WhereClause (boolean withValues, null);} should be sufficient maintain compatibility with existing code. Having instance variable in inconsistent state is always a potential source of unwanted side effect in future.

Reply:
@hengsin ohk Let me do this.